src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
I imagine that we might have a place that requires to have not only the options value, but also the API to operate with options (Get, Add, Remove etc). I also agree that this interface is backward-compatible.

But I think that this interface violates with [Interface Segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle). If I understand it right, methods (Get, Add, Remove) are required for Auth stuff. But for just retrieving options values isn't required.

Maybe create separated interface `IOptionsService` with methods (Get, Add, Remove). For basic use-cases  it in `IOptions` implementation. For Auth - inject the service itself and operate with options.